src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiSchemaGenerator.cs

```diff
@@ -118,6 +118,20 @@ internal static class OpenApiSchemaGenerator
                 AdditionalProperties = new OpenApiSchema { Type = "object" }
             };
 
+            if (context.Settings.EnablePagination)
+            {
+                schemas[Constants.CollectionPaginationResponse] = new()
+                {
+                    Title = "Base collection response",
+                    Type = "object",
+                    Properties = new Dictionary<string, OpenApiSchema>
+                    {
+                        { "@odata.nextLink", new OpenApiSchema { Type = "string"} },
```

Comment:
should we set them both as nullable? They are not always present

Reply:
sure, but pagination can be enabled from a translation perspective and the service still not reply with a next link and/or count at runtime.

Reply:
Resolved this with `nullable = true`